ticketbuyer/purchase.go

```diff
@@ -81,11 +92,30 @@ out:
 				case <-quit:
 					return
 				}
-				// Purchase tickets for each attached block, not just for the
-				// update to the main chain.  This is probably not optimal but
-				// it matches how dcrticketbuyer worked.
-				for h := v.NewHeight - int32(len(v.AttachedBlocks)) + 1; h <= v.NewHeight; h++ {
-					p.purchase(int64(h))
+
+				// fetch current peer block heights
+				peerInfo, err := p.w.ChainClient().GetPeerInfo()
+				peerBlockHeights := &Int64Slice{}
+				if err != nil {
+					log.Errorf("Failed to get peer info")
+				}
+
+				for _, peer := range peerInfo {
+					*peerBlockHeights = append(*peerBlockHeights, peer.CurrentHeight)
+				}
+
+				// get the median block height and try a ticket purchase
+				sort.Sort(*peerBlockHeights)
```

Comment:
Since go1.8 is a minimum now, I think this can/should all be shortened up with `sort.Slice`, using something like this leading into the median bits:

```go
				peerBlockHeights := make([]int64, len(peerInfo))
				for i := range peerInfo {
					peerBlockHeights[i] = peerInfo[i].CurrentHeight
				}

				// get the median block height and try a ticket purchase
				sort.Slice(peerBlockHeights, func(i, j int) bool {
					return peerBlockHeights[i] < peerBlockHeights[j]
				})
```

Basically you can get rid of `Int64Slice` entirely.  This seems to be an ideal case since the implementation of `sort.Interface` was almost entirely boilerplate, and it complicated matters down here (e.g. `Get`).